beets/dbcore/db.py

```diff
@@ -1123,9 +1144,14 @@ def bytelower(bytestring: Optional[AnyStr]) -> Optional[AnyStr]:
 
             return bytestring
 
-        conn.create_function("regexp", 2, regexp)
-        conn.create_function("unidecode", 1, unidecode)
-        conn.create_function("bytelower", 1, bytelower)
+        deterministic = {}
+        if sys.version_info >= (3, 8) and sqlite_version_info >= (3, 8, 3):
+            # Let sqlite make extra optimizations
+            deterministic["deterministic"] = True
```

Comment:
Consider using `partial` here, it may make it just slightly cleaner

```python
create_function = conn.create_function
...
    create_function = partial(create_function, deterministic=True)

create_function("regexp", 2, regexp)
...
```